src/query/storage/promremote/storage_test.go

```diff
@@ -426,6 +450,13 @@ func closeWithCheck(t *testing.T, c io.Closer) {
 	require.NoError(t, c.Close())
 }
 
+func verifyMetrics(t *testing.T, scope tally.TestScope) {
+	tallytest.AssertGaugeValue(
+		t, 0, scope.Snapshot(), "test_scope.prom_remote_storage.in_flight_samples",
```

Comment:
I wanna to make sure the in flight samples drain down to 0 after the store closes and it pass unit tests: @hczhu-db @yuchen-db